### PR TITLE
#157997115 Create a 404 page

### PR DESCRIPTION
**Pull request by kimotho-njoki**
## What does this PR do?
Introduces a page for all 404 errors.

## Description of Task to be completed?
* Add a PageNotFound component
* Add tests accordingly

## How should this be manually tested?
* Pull the branch
* Run `npm start` on your terminal
* Type in an incorrect route on the url

## What are the relevant pivotal tracker stories?
[#157997115](https://www.pivotaltracker.com/n/projects/2146417/stories/157997115)

## Any background context you want to add?
N/A

## Important notes
N/A

## Packages installed
N/A

## Deployment note
N/A

## Related PRs branch | PR (branch_name) | (pr_link)
Branch | PR
------ | ------
other_pr_production | [link]()
other_pr_master | [link]()
## Todos
- [x] Raise PR
- [ ] Test
- [ ] Documentation
## Screenshots
screen shot 2018-06-20 at 11 01 08

screen shot 2018-06-13 at 18 56 14

### CODE REVIEW

File: src/__test__/PageNotFoundComponent.test.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import expect from 'expect';
+
+import PageNotFoundComponent from '../components/PageNotFoundComponent.jsx';
+
+describe('PageNotFoundComponent is rendered when there is a 404 error', () => {
+
```
Comment (sachungo) [inline src/__test__/PageNotFoundComponent.test.js]: This extra space is not needed, please delete it

File: src/_css/PageNotFoundComponent.css [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+.ui.card>.content, .ui.cards>.card>.content {
+  padding-top: 30px;
+}
+
+.ui.card>.content p, .ui.cards>.card>.content p {
+  margin: 0px;
+  font-family: 'Frijole', cursive;
+  font-size: 5.6vw;
+  color: #fff;
+  /* color: black; */
```
Comment (sachungo) [inline src/_css/PageNotFoundComponent.css]: If not used, then delete

File: src/_css/PageNotFoundComponent.css [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+.ui.card {
+  box-shadow: 2px 2px 81px -8px rgba(0,0,0,0.65) !important;
+  background: none;
+  border-radius: 0px;
+}
+
+.ui.card>.content>.description,
+ .ui.cards>.card>.content>.description {
```
Comment (sachungo) [inline src/_css/PageNotFoundComponent.css]: The indentation is off. It should be aligned with line 46

File: src/_css/PageNotFoundComponent.css [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+   padding-top: 18px;
+   font-family: 'Mitr', sans-serif;
+   font-size: 0.6vw;
+   font-weight: normal !important;
+   color: rgba(0,0,0,.4);
+ }
+
+#andela-logo-404 {
+    position: absolute;
```
Comment (sachungo) [inline src/_css/PageNotFoundComponent.css]: Indentation for `#andela-logo-404` is off. It's supposed to be `2 spaces for each level of indentation` not `4 spaces`

**Comment by kimotho-njoki** [comment]
@sachungo feedback received. Working on it right away.

**Comment by kimotho-njoki** [comment]
@sachungo Still waiting on some feedback from @andela-hms on the design. Once that is done I will merge it.

**Comment by sachungo** [comment]
@kimotho-njoki have you gotten the design feedback yet?

**Comment by kimotho-njoki** [comment]
Hey @sachungo. Not yet. Haven't had a chance to sync with Humphrey but I'm setting up a call right now to do that. Will update you.

**Comment by kimotho-njoki** [comment]
@sachungo @andela-hms New designs incorporated. Please do give feedback.

**Comment by sachungo** [comment]
@kimotho-njoki 
- Fix the houndci comments. E.g.
> Unexpected usage of singlequote jsx-quotes

JSX prop values uses double quotes while strings uses single quotes. I.e.
```
const myString = 'I am a string'
....

<MyComponent  example="example prop" />
```
 
>`HTML entities must be escaped react/no-unescaped-entities`

add `"react/no-unescaped-entities": ["error", {"forbid": [">", "}"]}]` to `.eslintrc`

> react/prop-types comments

you can just use object proptype for history

- Fix the conflict

**Pull request merged**